### PR TITLE
[netcore] Minor fixes around System.Reflection

### PR DESCRIPTION
**Pull request by EgorBo**
Fixes https://github.com/mono/mono/issues/14909 (fixed by my previous PR)
Fixes https://github.com/mono/mono/issues/14950
Fixes https://github.com/mono/mono/issues/14951

### CODE REVIEW

File: netcore/sample/HelloWorld/HelloWorld.csproj [hunk 1]
```diff
@@ -4,6 +4,7 @@
     <OutputType>Exe</OutputType>
     <OutputPath>bin</OutputPath>
     <TargetFramework>netcoreapp3.0</TargetFramework>
+    <DebugType>full</DebugType>
```
Comment (akoeplinger) [inline netcore/sample/HelloWorld/HelloWorld.csproj]: what's the reason for this change? netcore should use portable pdb everywhere

Comment (EgorBo) [reply netcore/sample/HelloWorld/HelloWorld.csproj]: unwanted change, removed

**Comment by vargaz** [comment]
@monojenkins build failed
